### ISSUE TITLE
sorting decks on  https://pennydreadfulmagic.com/seasons/all/people/bakert99/ by record puts a 5-0 with suicide black too high up (above 6-0s) - why?

### ISSUE THREAD

**Issue by vorpal-buildbot** [PennyDreadfulMTG/Penny-Dreadful-Tools#7151]
Reported on Discord by bakert#2193

**Comment by bakert** [comment]
deck_cache thinks this deck has won 6 times. Why isn't the nightly updating it?

**Comment by bakert** [comment]
deck_id = 64180

**Comment by bakert** [comment]
deck.prime_cache doesn't actually alter wins/losses/draws. So once you are in a bad state in the cache you remain in a bad state forever?

**Comment by bakert** [comment]
The underlying issue here was probably that solved in #6790.

**Comment by bakert** [comment]
SQL from #5597:

```
SELECT dm.deck_id, COUNT(*) AS matches, wins + losses + draws AS total, wins, losses, draws  FROM deck_match AS dm INNER JOIN deck_cache AS dc ON dm.deck_id = dc.deck_id GROUP BY dm.deck_id HAVING COUNT(*) <> (dc.wins + dc.losses + dc.draws);
+---------+---------+-------+------+--------+-------+
| deck_id | matches | total | wins | losses | draws |
+---------+---------+-------+------+--------+-------+
|   48039 |       5 |     6 |    2 |      4 |     0 |
|   48043 |       3 |     4 |    0 |      4 |     0 |
|   64180 |       5 |     6 |    6 |      0 |     0 |
|   64185 |       5 |     6 |    2 |      4 |     0 |
|   68570 |       1 |     2 |    2 |      0 |     0 |
|   68582 |       1 |     2 |    0 |      2 |     0 |
+---------+---------+-------+------+--------+-------+
6 rows in set (0.23 sec)
````

**Comment by bakert** [comment]
Looking at the fix query from #5597 it would fix these six and also update the active_date of 155 other decks.

```
UPDATE
    deck_cache AS dc
LEFT JOIN
    (
        SELECT
            d.id,
            IFNULL(SUM(CASE WHEN dm.games > IFNULL(odm.games, 0) THEN 1 ELSE 0 END), 0) AS wins, -- IFNULL so we still count byes as wins.
            IFNULL(SUM(CASE WHEN dm.games < odm.games THEN 1 ELSE 0 END), 0) AS losses,
            IFNULL(SUM(CASE WHEN dm.games = odm.games THEN 1 ELSE 0 END), 0) AS draws,
            IFNULL(MAX(m.date), d.created_date) AS active_date
        FROM
            deck AS d
        LEFT JOIN
            deck_match AS dm ON dm.deck_id = d.id
        LEFT JOIN
            deck_match AS odm ON dm.match_id = odm.match_id AND dm.deck_id <> odm.deck_id
        LEFT JOIN
            `match` AS m ON m.id = dm.match_id
        GROUP BY
            d.id
    ) AS dsum ON dc.deck_id = dsum.id
SET
    dc.wins = dsum.wins,
    dc.draws = dsum.draws,
    dc.losses = dsum.losses,
    dc.active_date = dsum.active_date;
```

This makes me realize our past fixes to deck_cache wins/lossess/draws don't update active_date and probably should. I've opened #7185  for that because it's mostly unrelated.

I've updated using this "lesser" query for now that just does those six wins/losses/draws:

```
UPDATE
    deck_cache AS dc
LEFT JOIN
    (
        SELECT
            d.id,
            IFNULL(SUM(CASE WHEN dm.games > IFNULL(odm.games, 0) THEN 1 ELSE 0 END), 0) AS wins, -- IFNULL so we still count byes as wins.
            IFNULL(SUM(CASE WHEN dm.games < odm.games THEN 1 ELSE 0 END), 0) AS losses,
            IFNULL(SUM(CASE WHEN dm.games = odm.games THEN 1 ELSE 0 END), 0) AS draws,
            IFNULL(MAX(m.date), d.created_date) AS active_date
        FROM
            deck AS d
        LEFT JOIN
            deck_match AS dm ON dm.deck_id = d.id
        LEFT JOIN
            deck_match AS odm ON dm.match_id = odm.match_id AND dm.deck_id <> odm.deck_id
        LEFT JOIN
            `match` AS m ON m.id = dm.match_id
        GROUP BY
            d.id
    ) AS dsum ON dc.deck_id = dsum.id
SET
    dc.wins = dsum.wins,
    dc.draws = dsum.draws,
    dc.losses = dsum.losses;
```
